Automattic/woocommerce-payments
# Fix Sofort transactions details display

Fixes #1669

#### Changes proposed in this Pull Request


### Transaction Detail Trauma



Previously if a transaction was completed using Sofort as a payment method, if you tried to view this transaction's details in the WCPay Admin (**Payments > Transactions**), you would only be able to view a disappointing empty screen. This is because this details page was expecting the transaction to be a credit card transaction and looking for card-specific fields that do not apply to Sofort transactions.

The changes in this PR remedy this situation by parsing the correct transaction details relevant to a Sofort transaction. 





#### Testing instructions





1. To start with, we must have at least one Sofort transaction in our site's list of transactions. For the necessary steps to enable Sofort and the requirements to complete a checkout using Sofort, please refer to the instructions in #1624 .
2. Before pulling these changes, if you select a transaction completed using Sofort, you should not be able to see any details.
3. After pulling these changes, please select a transaction from your list of transactions that uses Sofort. In the detail summary, near the top of the page, you should now see the correct information, including the last four digits of a censored IBAN alongside a pink Sofort logo. At the bottom of the page, you should see the full list of transaction details, under the section titled `Payment method`. If you go into your site's Stripe dashboard and find this transaction amongst the list of payments, the details should all match up. Please note that the fields presented and the order in which they are displayed in WCPay reflects the way are presented in Stripe's dashboard.
4. To confirm that these changes do not conflict with any existing functionality, please check the transaction details for credit cards/SEPA/Giropay and confirm that the relevant fields are still present as expected.

-------------------
